packages/amplication-server/src/core/block/block.service.ts

```diff
@@ -942,4 +942,35 @@ export class BlockService {
       },
     });
   }
+
+  async getLatestVersions(args: {
+    where: Prisma.BlockWhereInput;
+  }): Promise<Block[]> {
+    const blocks = await this.prisma.block.findMany({
+      where: {
+        ...args.where,
+        resourceId: args.where.resourceId,
+        deletedAt: null,
+      },
+      include: {
+        versions: {
+          where: {
+            versionNumber: {
+              not: CURRENT_VERSION_NUMBER,
+            },
+          },
+          take: 1,
+          orderBy: {
+            versionNumber: Prisma.SortOrder.desc,
+          },
+        },
+        parentBlock: true,
+      },
+    });
+
+    return (await blocks).map((block) => {
```

Comment:
Why do you need to await blocks again? you already await the findMany call above